raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py

```diff
@@ -155,24 +222,67 @@ def test_remote_transaction_with_zero_gasprice_is_not_mined(
         raise RuntimeError(f"Unknown blockchain_type {blockchain_type}")
 
 
-def test_duplicated_transaction_different_gas_price_raises(deploy_client: JSONRPCClient) -> None:
+def test_resending_pending_transaction_with_lower_gas_raises(deploy_client: JSONRPCClient) -> None:
     """ If the same transaction is sent twice a JSON RPC error is raised. """
-    gas_price = GasPrice(2000000000)
-    deploy_client.web3.eth.setGasPriceStrategy(make_decreasing_gas_price_strategy(gas_price))
+    # Use a _decreasing_ gas price startegy so that the second transactions is
+    # lower than the first.
+    deploy_client.web3.eth.setGasPriceStrategy(
+        make_decreasing_gas_price_strategy(GasPrice(2000000000))
+    )
     contract_proxy, _ = deploy_rpc_test_contract(deploy_client, "RpcTest")
 
     address = contract_proxy.contract_address
     assert len(deploy_client.web3.eth.getCode(to_checksum_address(address))) > 0
 
-    second_client = JSONRPCClient(web3=deploy_client.web3, privkey=deploy_client.privkey)
+    client_invalid_nonce = JSONRPCClient(web3=deploy_client.web3, privkey=deploy_client.privkey)
+
+    check_block = deploy_client.get_checking_block()
+    gas_estimate = contract_proxy.estimate_gas(check_block, "ret")
+    assert gas_estimate, "Gas estimation should not fail here"
+    startgas = safe_gas_limit(gas_estimate)
+
+    contract_proxy.transact("ret", startgas)
+
+    # At this point `client_invalid_nonce` has a nonce that is `1` too low,
+    # since a transaction was sent using `deploy_client` above and these two
+    # instances share the same underlying private key.
+    #
+    # Note that a the same function is called in this test but the gas is decreasing.
+    with pytest.raises(ReplacementTransactionUnderpriced):
+        client_invalid_nonce.new_contract_proxy(
+            abi=contract_proxy.contract.abi, contract_address=contract_proxy.contract_address
+        ).transact("ret", startgas)
+
 
-    second_proxy = second_client.new_contract_proxy(
-        abi=contract_proxy.contract.abi, contract_address=contract_proxy.contract_address
+def test_reusing_nonce_with_lower_gas_raises(deploy_client: JSONRPCClient) -> None:
```

Comment:
This was the initial test added by this PR, I moved from it together to the other 4 tests.